### ISSUE TITLE
[app] no response on action

### ISSUE THREAD

**Issue by florianlenz** [Bit-Nation/BITNATION-Pangea-mobile#389]
## Feature / Issue
As a user of the app I would like to get feedback about my actions as soon as possible. When I press buttons etc it happens that they response 10 seconds later. Since I am a very impatient human beeing I want to see the result 1ms later OR a indication of that there is something happen so that I know "ok - the app is processing my action". If I stress test the app it happens that there are 40+ screens pushed into the navigation. We need to fix that in the whole app. 

## Acceptance criteria
- [ ] I just want to get result when I trigger a button

## Additional information
[Here is a video](https://vimeo.com/274314987) you can't see what I am doing, but I am pressing a lot of times the send / receive button in wallet and all the buttons in the settings page.

**Comment by florianlenz** [comment]
I am also not able to get into a chat for sever seconds. I am very inpatient!

**Comment by seland** [comment]
As far as I can imagine, that happens because we have too much work happening on JS thread, specifically nation indexing is quite hard operation (a lot of blocks received, processed and saved to database). Also, as written [here](https://github.com/realm/realm-js/issues/1002) Realm writes are heavyweight operations themselves, but we currently do 1-2 writes for every nation while indexing. 
So, for anyone who will investigate this, I suggest to start from here.

Another issue is that we have console.log with possibly large object like socket error messages and others. I would consider some tool that will prevent logging to console in release builds.

**Comment by florianlenz** [comment]
hmh I see. I guess we should then refactor the code to index them in memory and persist them when we are done? Reads are no problem, right? 

> Another issue is that we have console.log with possibly large object like socket error messages and others. I would consider some tool that will prevent logging to console in release builds.

I think `console.log` is removed in production builds. It's useless anyway afaik.

**Comment by florianlenz** [comment]
[This](https://facebook.github.io/react-native/docs/interactionmanager.html) might be useful as well.

**Comment by florianlenz** [comment]
We should also check this list: https://facebook.github.io/react-native/docs/performance.html

**Comment by infinitechaos** [comment]
I am running an LG V20 and have also experienced issues with the app's actions lagging. For example - I just tapped 'share wallet address' and nothing happened. So I tapped it another 3-4 times and still nothing happened. So I set my phone down and a full two minutes later the 'share with' menu finally popped up. I selected 'copy to clipboard' and then the 'share with' menu proceeded to show up once again for every time I had tapped the 'share wallet address' button.

I experienced this same delay and lag when I was setting up my wallet/app last week, backing up the private key, etc.
